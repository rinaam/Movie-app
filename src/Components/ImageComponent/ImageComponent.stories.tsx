import React from "react";
import { Story, Meta } from "@storybook/react";

import ImageComponent, { IImageProps } from "./ImageComponent";
import { ImageSizes } from "../../Constants";

export default {
  title: "Image",
  component: ImageComponent,
  argTypes: {
    size: {
      control: {
        type: "radio",
        options: [
          ImageSizes.xs,
          ImageSizes.s,
          ImageSizes.m,
          ImageSizes.l,
          ImageSizes.xl,
          ImageSizes.xxl,
          ImageSizes.xxl,
          ImageSizes.original,
        ],
      },
    },
  },
} as Meta;

const Template: Story<IImageProps> = (args) => <ImageComponent {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  imgUrl: `https://image.tmdb.org/t/p/w185/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg`,
  alt: "image",
};

const TemplateSizes: Story<{ size: ImageSizes }> = (args) => {
  return (
    <ImageComponent
      imgUrl={`https://image.tmdb.org/t/p/${args.size}/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg`}
      alt="image"
    />
  );
};

export const CustomSizes = TemplateSizes.bind({});
CustomSizes.args = {
  size: ImageSizes.l,
};
