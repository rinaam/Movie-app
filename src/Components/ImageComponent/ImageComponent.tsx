import React from "react";

export interface IImageProps {
  imgUrl: string;
  alt: string;
  useBackground?: boolean;
  className?: string;
}

const ImageComponent: React.FC<IImageProps> = ({
  imgUrl,
  alt,
  useBackground = false,
  className = "",
}) =>
  useBackground ? (
    <div
      className={className}
      style={{
        backgroundImage: `url("${imgUrl}")`,
      }}
    />
  ) : (
    <img className={className} src={imgUrl} alt={alt} />
  );

export default ImageComponent;
