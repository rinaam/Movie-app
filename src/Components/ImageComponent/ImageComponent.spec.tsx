import React from "react";
import { create } from "react-test-renderer";
import ImageComponent from "./ImageComponent";

describe("ImageComponent", () => {
  it("should render ImageComponent", () => {
    const wrapper = create(<ImageComponent imgUrl="fakeUrl" alt="fakeAlt" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it("should render ImageComponent with a background", () => {
    const wrapper = create(
      <ImageComponent imgUrl="fakeUrl" alt="fakeAlt" useBackground />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it("should accept classname", () => {
    const wrapper = create(
      <ImageComponent
        imgUrl="fakeUrl"
        alt="fakeAlt"
        useBackground
        className="fakeClassname"
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
