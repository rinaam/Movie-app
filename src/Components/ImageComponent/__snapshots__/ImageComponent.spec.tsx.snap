// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageComponent should accept classname 1`] = `
<div
  className="fakeClassname"
  style={
    Object {
      "backgroundImage": "url(\\"fakeUrl\\")",
    }
  }
/>
`;

exports[`ImageComponent should render ImageComponent 1`] = `
<img
  alt="fakeAlt"
  className=""
  src="fakeUrl"
/>
`;

exports[`ImageComponent should render ImageComponent with a background 1`] = `
<div
  className=""
  style={
    Object {
      "backgroundImage": "url(\\"fakeUrl\\")",
    }
  }
/>
`;
